### PR TITLE
/signup UI & functionality

### LINKED ISSUE #195: UI for /signup
Tentative UI for the Sign up page.

### PR DESCRIPTION
**Pull request by cypherhoodlum**
### What does it do? Why?

Closes #195 and #196.

- UI for /signup
- nostr profile creation
- logs in the created user with their **pubkey**

Screenshot from 2023-03-13 21-28-41

Screenshot from 2023-03-13 21-29-10

### Review

- All GHA are success
- Use https://ui.shadcn.com/docs often as possible
- If Vercel build fails, run `yarn build` locally to get the same errors
      more to come...

### CODE REVIEW

File: src/app/signup/page.tsx [hunk 1]
```diff
@@ -1,17 +1,134 @@
...
+            </div>
+          </div>
+          {copied && <div className="flex justify-center mt-2">
+            <span className="text-xs">Copied to clipboard.</span>
+          </div>}
+        </div>
       </div>
-    </div>
+      {showMsg &&
+        <div className="bg-[#171B21] py-6 flex flex-col items-center">
```
Comment (PeerRich) [inline src/app/signup/page.tsx:124]:
```suggestion
        <div className="bg-[#171B21] py-6 px-4 flex flex-col items-center">
```

**Comment by PeerRich** [comment]
nice the generator works, I just wonder if this UX is what we want. right now you just click on "generate" and its not quite clear what happens. 

other nostr clients log you in immediately etc.

**Comment by cypherhoodlum** [comment]
> other nostr clients log you in immediately etc.

That's one option, sure. I just see people losing their keys really easily if it's not something they interact with themselves straight from the start. I don't know, do you think we should just log the user in immediately and have them backup the key from the settings account page instead? Signup would just tell the user to do that?

**Comment by cypherhoodlum** [comment]
> wouldn't that force to store the user's private key to localStorage?

Yes. Right now when you press 'generate keys' the user is logged in with the public key. It is up to the user to backup the private key immediately. Maybe the user should also confirm they have backed it up before leaving the page?

**Comment by PeerRich** [comment]
Maybe we just make change the button after clicking to "Sign in" that opens a new tab with the sign in page

**Comment by PeerRich** [comment]
thoughts on this? if you like this I can commit it

https://user-images.githubusercontent.com/8019099/225000671-6d3417c4-686c-4175-9304-80c363e9b218.mp4

**Comment by cypherhoodlum** [comment]
Looks good!

**Comment by PeerRich** [comment]
pushed

**Comment by PeerRich** [comment]
> Looks good! edit: What happens if the user forgets to copy the private key, tho?

since it opens in a new tab its quite save

**Review by PeerRich — APPROVED** [review]

**Comment by PeerRich** [comment]
actually fine to merge, any follow up changes can be done in a new PR

**Comment by PeerRich** [comment]
excited for this!

**Pull request merged**